File: Gemini/Simples/dfs/dfs.py
#Input: Using the previous graph, how to implement the Depth-First Search algorith?

graph = {
    'A': ['B', 'C'],
    'B': ['D', 'E'],
    'C': [],
    'D': [],
    'E': []
}

def dfs(graph, start):
    """
    Performs Depth-First Search on the given graph starting from the specified vertex.

    Args:
        graph: The graph represented as a dictionary.
        start: The starting vertex.

    Returns:
        A list containing the visited vertices in DFS order.
    """

    visited = set()  # Set to keep track of visited vertices
    stack = [start]  # Stack to store vertices for exploration

    while stack:
        node = stack.pop()
        if node not in visited:
            visited.add(node)
            stack.extend(reversed(graph[node]))  # Add neighbors in reverse order for DFS

    return list(visited)


result = dfs(graph, 'A')
print(result)

"""
Output: ['A', 'D', 'C', 'E', 'B']
Wrong:
Should be: ['A', 'B', 'D', 'E', 'C']
Score: 0.5
"""

File: Gemini/Simples/dfs/Dfs.java
// Input  Using the previous graph, how to implement the Depth-First Search algorith?

public void dfs(int startVertex) {
    boolean[] visited = new boolean[vertices.size()];
    dfsHelper(startVertex, visited);
}

private void dfsHelper(int vertex, boolean[] visited) {
    visited[vertex] = true;
    System.out.print(vertices.get(vertex).getLabel() + " "); // Process vertex

    for (Vertex neighbor : vertices.get(vertex).getAdjacentVertices()) {
        if (!visited[neighbor.getLabel()]) {
            dfsHelper(neighbor.getLabel(), visited);
        }
    }
}

/*

Error in Compilation 
Score: 0

*/